js/index.js

```diff
@@ -1,9 +1,353 @@
+  //顔を回答するモード
+  let faceMode = false;
+  //表示インデックス
+  let randomIndex;
+  //問題数を格納 最終問題かの判定を行う
+  const questionLength = sysmember.length;
+  //何問表示したかをカウント
+  let count = 0;
+  //結果格納 
+  let result;
+  //正解数カウント
+  let correctCount = 0;
+  //正答格納用
+  let correctAnswer;
+  //正解率
+  let accuracyRate; 
+  //アニメーション
+  let animation;
+
+  const gameControl = document.getElementById('gameControl'); // 開始ボタン
+  const image = document.getElementById('icon'); // ロゴ画像
+  
+  // スタートボタンをクリックしたらアイコンが動き出す
+  gameControl.addEventListener('click', () => {
+    if (gameControl.textContent === 'START') {
+      startAnimation();
+      //１回目の抽選
+      setQuestion();
+      //ボタン表示変更
+      gameControl.textContent = 'OK';
+      //成績欄を非表示
+      document.getElementById('performance').style.display = "none";
+      return;
+    };
+    if (gameControl.textContent === 'OK') {
+      let answer = checkAnswer();
+      if(answer === undefined){
+        alert("選択してください");
+        return;
+      };
+      result = judge(answer);
+      displaydialog(result);
+    };
+  });
+
+  //アニメーションを開始する関数
+  function startAnimation() {
+    animation = image.animate(
+      // 途中の状態を表す配列
+      [
+        { transform: 'translateX(0)' }, // 開始時の状態（左端）
+        { transform: 'translateX(1500px)' } // 終了時の状態（左端から200pxの位置）
+      ],
+      // タイミングに関する設定
+      {
+        fill: 'forwards', // 再生後、最後の状態を保持
+        duration: 60000, // 再生時間（1000ミリ秒）
+      },
+    );
+  }
+
+  //アニメーションを停止する関数
+  function stopAnimation() {
+    if (animation) {
+      animation.pause(); //一時停止
+    }
+  }
+
+  //表示する問題をランダムに抽選する関数
+  function getrandomIndex() {
+    randomIndex = Math.floor(Math.random() * sysmember.length);
+    return sysmember[randomIndex];
+  };
+
+  //問題セット関数
+  function setQuestion() {
+    if (nicknameMode) {
+      const questionImage = document.getElementById('question-image');
+      questionImage.src = getrandomIndex().src;
+    };
+    if (faceMode) {
+      const questionLabel = document.getElementById('question-label');
+      questionLabel.textContent = getrandomIndex().nickname;
+    };
+    correctAnswer = sysmember[randomIndex].value;
+    sysmember.splice(randomIndex, 1);
+    count += 1;
+  };
+
+  //回答で何が選択されているかチェックする関数
+  function checkAnswer() {
+    const nicknameOption = document.getElementsByName("nickname-option");
+    const faceOption = document.getElementsByName("face-option");
+    if (nicknameMode) {
+      for (let i = 0; i < nicknameOption.length; i++) {
+        if (nicknameOption[i].checked) {
+          return nicknameOption[i].value;
+        };
+      };
+    };
+    if (faceMode) {
+      for (let i = 0; i < faceOption.length; i++) {
+        if (faceOption[i].checked) {
+          return faceOption[i].value;
+        };
+      };
+    };
+  };
+  //回答の正誤判定関数
+  function judge(Useranswer) {
+    if (correctAnswer === Useranswer) {
+      return result = true;
+    }
+    else {
+      return result = false;
+    };
+  };
```

Comment:
ここで`result`に結果を入れても、この関数の呼び出し元で`result`に再代入しているみたいなので、
ここで`result`に結果を入れる必要はないかも？
あと、else句を書かなくても下記で同じ結果を生み出せますので覚えておいてください～。
```suggestion
  function judge(Useranswer) {
    if (correctAnswer === Useranswer) {
      return true;
    }
    return false
  };
```